Library/Homebrew/test/utils_spec.rb

```diff
@@ -0,0 +1,293 @@
+require "utils"
+
+describe self do
+  let(:dir) { @dir = Pathname.new(Dir.mktmpdir) }
+
+  after(:each) do
+    dir.rmtree unless @dir.nil?
+  end
+
+  def esc(code)
+    /(\e\[\d+m)*\e\[#{code}m/
+  end
+
+  describe "#ofail" do
+    it "sets Homebrew.failed to true" do
+      shutup do
+        ofail "foo"
+      end
+
+      expect(Homebrew).to have_failed
+    end
+  end
+
+  describe "#odie" do
+    it "exits with 1" do
+      expect(self).to receive(:exit).and_return(1)
+      shutup do
+        odie "foo"
+      end
+    end
+  end
+
+  describe "#pretty_installed" do
+    context "when $stdout is a TTY" do
+      before(:each) do
+        allow($stdout).to receive(:tty?).and_return(true)
+      end
+
+      context "with HOMEBREW_NO_EMOJI unset" do
+        before(:each) do
+          ENV.delete("HOMEBREW_NO_EMOJI")
+        end
+
+        it "returns a string with a colored checkmark" do
+          expect(pretty_installed("foo"))
```

Comment:
Perhaps we should set a subject here, since we're always testing the same value?